### PR TITLE
[wasm] Misc tests related fixes

### LINKED ISSUE #73872: Test failure: WasmTestOnBrowser-subtlecrypto-System.Security.Cryptography.Tests
WasmTestOnBrowser-subtlecrypto-System.Security.Cryptography.Tests failure found in https://github.com/dotnet/runtime/pull/73748/checks?check_run_id=7812609951

[LOG](https://helix.dot.net/api/2019-06-17/jobs/66ea24e6-d195-4f66-9a72-0e2ebe9b298c/workitems/WasmTestOnBrowser-subtlecrypto-System.Security.Cryptography.Tests/console): 
```log
[19:16:18] fail: MONO_WASM_ENCRYPT_DECRYPT: Error: Worker failed
    at us.acquire_lock (http://127.0.0.1:49196/dotnet.js:3:63935)
    at using_lock (http://127.0.0.1:49196/dotnet.js:3:63647)
    at us.reset (http://127.0.0.1:49196/dotnet.js:3:61805)
    at us.send_msg (http://127.0.0.1:49196/dotnet.js:3:61114)
    at cs (http://127.0.0.1:49196/dotnet.js:3:59745)
    at is (http://127.0.0.1:49196/dotnet.js:3:58724)
    at Object.ts (http://127.0.0.1:49196/dotnet.js:3:57718)
    at _dotnet_browser_simple_digest_hash (http://127.0.0.1:49196/dotnet.js:14:97296)
    at SystemCryptoNativeBrowser_SimpleDigestHash (http://127.0.0.1:49196/dotnet.wasm:wasm-function[8951]:0x1d372f)
    at do_icall (http://127.0.0.1:49196/dotnet.wasm:wasm-function[355]:0x22d31)
[19:16:18] fail: [FAIL] System.Security.Cryptography.Tests.Sha1Tests.Sha1_Rfc3174_3_Async
[19:16:18] info: System.Security.Cryptography.CryptographicException : SubtleCrypto returned an unknown error: '-3'.
[19:16:18] info:    at System.Security.Cryptography.SHANativeHashProvider.SimpleDigestHash(SimpleDigest hashName, ReadOnlySpan`1 data, Span`1 destination)
[19:16:18] info:    at System.Security.Cryptography.SHANativeHashProvider.GetCurrentHash(Span`1 destination)
[19:16:18] info:    at System.Security.Cryptography.SHANativeHashProvider.FinalizeHashAndReset(Span`1 destination)
[19:16:18] info:    at System.Security.Cryptography.LiteHash.Finalize(Span`1 destination)
[19:16:18] info:    at System.Security.Cryptography.LiteHashProvider.<ProcessStreamAsync>d__9`1[[System.Security.Cryptography.LiteHash, System.Security.Cryptography, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].MoveNext()
[19:16:18] info:    at System.Security.Cryptography.LiteHashProvider.<ProcessStreamAsync>d__10`1[[System.Security.Cryptography.LiteHash, System.Security.Cryptography, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].MoveNext()
[19:16:18] info:    at System.Security.Cryptography.Tests.HashAlgorithmTestDriver.VerifyOneShotAllocatingStreamAsync(Stream input, String output)
[19:16:18] info:    at System.Security.Cryptography.Tests.HashAlgorithmTestDriver.VerifyRepeatingAsync(String input, Int32 repeatCount, String output)
[19:16:18] info:    at System.Security.Cryptography.Tests.Sha1Tests.Sha1_Rfc3174_3_Async()
[19:16:18] info: --- End of stack trace from previous location ---
[19:16:18] fail: MONO_WASM_ENCRYPT_DECRYPT: Error: Worker failed
    at us.acquire_lock (http://127.0.0.1:49196/dotnet.js:3:63935)
    at using_lock (http://127.0.0.1:49196/dotnet.js:3:63647)
    at us.reset (http://127.0.0.1:49196/dotnet.js:3:61805)
    at us.send_msg (http://127.0.0.1:49196/dotnet.js:3:61114)
    at cs (http://127.0.0.1:49196/dotnet.js:3:59745)
    at is (http://127.0.0.1:49196/dotnet.js:3:58724)
    at Object.ts (http://127.0.0.1:49196/dotnet.js:3:57718)
    at _dotnet_browser_simple_digest_hash (http://127.0.0.1:49196/dotnet.js:14:97296)
    at SystemCryptoNativeBrowser_SimpleDigestHash (http://127.0.0.1:49196/dotnet.wasm:wasm-function[8951]:0x1d372f)
    at do_icall (http://127.0.0.1:49196/dotnet.wasm:wasm-function[355]:0x22d31)
[19:16:18] fail: [FAIL] System.Security.Cryptography.Tests.Sha1Tests.ComputeHash_TryComputeHash_HashSetExplicitlyByBoth
...
```

### LINKED ISSUE #73901: [wasm] Tests failing to build with trimming
[Rolling build](https://dev.azure.com/dnceng/public/_build/results?buildId=1941593&view=logs&j=06c2b9cd-9a25-5ea1-8363-c1b61a607e70&t=07e56740-dd3a-5379-9df5-8d02f9b6c188):

`IL2121`, and `IL2104`:
```
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(379,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, String, String, String): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(425,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, String, Byte[]): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(470,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, Int32, Int32, Int32): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(510,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, Int32, String): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(547,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, String, Int32): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(584,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, String, String, Int32): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(628,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, Int32, Int32, Int32, Int32, Int32, Int32, Int32): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 
/_/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs(688,13): Trim analysis error IL2121: System.Net.NetEventSource.WriteEvent(Int32, String, String, Int32, Int32, Int32): Unused 'UnconditionalSuppressMessageAttribute' for warning 'IL2026'. Consider removing the unused warning suppression. 

/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.ComponentModel.TypeConverter.dll : error IL2104: Assembly 'System.ComponentModel.TypeConverter' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.Linq.Expressions.dll : error IL2104: Assembly 'System.Linq.Expressions' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.Runtime.InteropServices.JavaScript.dll : error IL2104: Assembly 'System.Runtime.InteropServices.JavaScript' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/native/System.Private.CoreLib.dll : error IL2104: Assembly 'System.Private.CoreLib' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/.dotnet/sdk/7.0.100-preview.5.22307.18/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.ILLink.targets(110,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false. 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.ComponentModel.TypeConverter.dll : error IL2104: Assembly 'System.ComponentModel.TypeConverter' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.Linq.Expressions.dll : error IL2104: Assembly 'System.Linq.Expressions' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/lib/net7.0/System.Runtime.InteropServices.JavaScript.dll : error IL2104: Assembly 'System.Runtime.InteropServices.JavaScript' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm/Release/runtimes/browser-wasm/native/System.Private.CoreLib.dll : error IL2104: Assembly 'System.Private.CoreLib' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries 
```

### PR DESCRIPTION
**Pull request by radical**
- Disable subtlecrypto tests
- Improve build times for runtime packs, in case of Wasm.Build.Tests
- Fix incremental builds for the above
- Fix running samples on threaded builds
- Don't run only smoke tests for EAT. This will help in catching trimming errors
- Fix trimming errors with tests

Fixes https://github.com/dotnet/runtime/issues/73872
Fixes https://github.com/dotnet/runtime/issues/73901

### CODE REVIEW

**Comment by radical** [comment]
I want to land this for 7.0 .

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ private void CollectPInvokes(List<PInvoke> pinvokes, List<PInvokeCallback> callb
             }
             catch (Exception ex) when (ex is not LogAsErrorException)
             {
-                Log.LogWarning(null, "WASM0001", "", "", 0, 0, 0, 0, $"Could not get pinvoke, or callbacks for method '{method.Name}' because '{ex.Message}'");
+                Log.LogWarning(null, "WASM0001", "", "", 0, 0, 0, 0, $"Could not get pinvoke, or callbacks for method '{type.FullName}::{method.Name}' because '{ex.Message}'");
```
Comment (maraf) [inline src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:74]: A similar message is also here https://github.com/dotnet/runtime/blob/main/src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs#L304 and in the icall generator https://github.com/dotnet/runtime/blob/main/src/tasks/WasmAppBuilder/IcallTableGenerator.cs#L155

File: src/libraries/System.Security.Cryptography/tests/System.Security.Cryptography.Tests.csproj [hunk 2]
```diff
@@ -14,8 +14,6 @@
     <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
-    <HelixTargetsFile>wasm.helix.targets</HelixTargetsFile>
```
Comment (eerhardt) [inline src/libraries/System.Security.Cryptography/tests/System.Security.Cryptography.Tests.csproj:17]: Can we delete the wasm.helix.targets file as well?

Comment (radical) [reply src/libraries/System.Security.Cryptography/tests/System.Security.Cryptography.Tests.csproj:17]: I moved the file to eng/testing as a sample.

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs (hunk 1 above)
Comment (radical) [reply src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs]: Fixed

**Comment by radical** [comment]
This fixes trimming errors on `main` too.

**Review by lambdageek — APPROVED** [review]
(primarily looked at the tasks, but the rest also makes sense to me)

**Comment by lambdageek** [comment]
```
Starting:    System.Runtime.Serialization.Json.ReflectionOnly.Tests.dll
[19:15:51] fail: [FAIL] DataContractJsonSerializerTests.DCJS_VerifyDateTimeForFormatStringDCJsonSerSettings
[19:15:51] info: System.InvalidOperationException : No set method for property 'UtcDateTime' in type 'System.Runtime.Serialization.DateTimeOffsetAdapter'.
[19:15:51] info:    at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
[19:15:51] info:    at DataContractJsonSerializerTests.SerializeAndDeserialize[DateTimeOffset](DateTimeOffset value, String baseline, DataContractJsonSerializerSettings settings, Func`1 serializerFactory, Boolean skipStringCompare)
[19:15:51] info:    at System.Reflection.MethodInvoker.InterpretedInvoke(Object , Span`1 , BindingFlags )
[19:15:52] fail: [FAIL] DataContractJsonSerializerTests.DCJS_ClassWithDatetimeOffsetTypeProperty
[19:15:52] info: System.InvalidOperationException : No set method for property 'UtcDateTime' in type 'System.Runtime.Serialization.DateTimeOffsetAdapter'.
[19:15:52] info:    at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
[19:15:52] info:    at DataContractJsonSerializerTests.SerializeAndDeserialize[TypeWithDateTimeOffsetTypeProperty](TypeWithDateTimeOffsetTypeProperty value, String baseline, DataContractJsonSerializerSettings settings, Func`1 serializerFactory, Boolean skipStringCompare)
[19:15:52] info:    at System.Reflection.MethodInvoker.InterpretedInvoke(Object , Span`1 , BindingFlags )
[19:15:52] info: Finished:    System.Runtime.Serialization.Json.ReflectionOnly.Tests.dll
```

haven't seen these before

**Comment by radical** [comment]
The failures are caused because:
1. `main` has trimming errors, which we fixed here (https://github.com/dotnet/runtime/issues/73901)
2. And we are now (as of this PR) running all the EAT tests, which then reveals the errors that you mentioned caused by trimming.

**Comment by radical** [comment]
There is a new linker update PR open (https://github.com/dotnet/runtime/pull/73894), and this might be better solved over there.
So, w.r.t the linker issues:
- this PR makes the situation slightly better but doesn't fully fix it.
- does not make it worse!

I think we can merge this. And I'll follow up in the other PR to fix all the linker issues.

**Pull request merged**